front-end/src/configuration/settings.jsx

```diff
@@ -70,6 +73,45 @@ class settings extends React.Component {
     }.bind(this)
   }
 
+  handleSetAddress (ev) {
+    const settings = this.state.settings
+    settings.address = ev.target.value
+    this.setState({
+      settings: settings,
+      updated: true
+    })
+  }
+
+  handleSetProtocolHttp () {
+    this.handleSetProtocol(false)
+  }
+
+  handleSetProtocolHttps () {
+    this.handleSetProtocol(true)
+  }
+
+  handleSetProtocol (value) {
```

Comment:
Is it possible to add a test for this?

Reply:
Good point. I'll work on adding tests for this tab tonight.

Reply:
A few tests have been added for the new functionality.